### PR TITLE
Introduce persistence milestoning library logical plan.

### PR DESCRIPTION
**Pull request by davidsunglee**
#### What type of PR is this?
Feature.

#### What does this PR do / why is it needed ?
We are introducing a common library to handle batch ingestion logic with an initial focus on batch milestoning into relational databases. This library consolidates the implementation across multiple backend persistence implementations. It also provides an in-memory implementation that will be of use for fast testing.

This PR introduces the first component of the solution above: logical plan.

#### Which issue(s) this PR fixes:
None. But this is the first of a series of PRs addressing #920

#### Does this PR introduce a user-facing change?
No.

### CODE REVIEW

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/pom.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>org.finos.legend.engine</groupId>
+        <artifactId>legend-engine-xt-persistence-component</artifactId>
+        <version>3.15.4-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>legend-engine-xt-persistence-component-logical-plan</artifactId>
```
Comment (rafaelbey) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/pom.xml:27]: Wonder if all packages should be prefixed with 'logical plan' to ensure no conflicts with future components?

Comment (davidsunglee) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/pom.xml:27]: Makes sense to have a prefix for namespacing.

'component' is too generic, so we'll replace that with something more descriptive (e.g. ingestion, milestoning, etc).

Will do it in one shot, once all related modules are pushed (targeting end of the week).

Comment (davidsunglee) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/pom.xml:27]: And your instinct is right, we'll have a lineage component in the near future.

**Review by rafaelbey — APPROVED** [review]

**Comment by davidsunglee** [comment]
@rafaelbey @epsstan Can you merge when you have a moment? I don't have authorization ... thanks!

**Pull request merged**